service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OthersControllerAboutToSubmitTest.java

```diff
@@ -88,18 +88,36 @@ void shouldNotSaveOtherWhenAllOtherElementsAreNullOrEmpty() {
         assertThat(response.getData()).doesNotContainKey("others");
     }
 
+    @Test
+    void shouldNotSaveConfidentialAddressWhenAddressIsNotKnown() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of("others", Others.builder()
+                .firstOther(other())
+                .additionalOthers(additionalOthersWithoutAddress())
+                .build()))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToSubmitEvent(caseDetails);
+        CaseData caseData = mapper.convertValue(response.getData(), CaseData.class);
+
+        assertThat(unwrapElements(caseData.getConfidentialOthers())).contains(confidentialOther());
+        assertThat(unwrapElements(caseData.getConfidentialOthers())).contains(confidentialOtherWithoutAddress());
```

Comment:
The .contains(confidentialOtherWithoutAddress()) can be chained to the line above

Reply:
changed